src/main/java/org/primeframework/mvc/cors/CORSConfiguration.java

```diff
@@ -98,8 +105,40 @@ public CORSConfiguration withDebug(boolean debug) {
     return this;
   }
 
+  private void checkExclusiveMatching() {
+    if (excludedPathPattern != null && includedPathPattern != null) {
+      throw new RuntimeException("You cannot use both withExcludedPathPattern and withIncludedPathPattern. Must be one or the other.");
+    }
+    if (includeUriChecker != null && excludeUriChecker != null) {
+      throw new RuntimeException("You cannot use both withIncludedUriChecker and withExcludedUriChecker. Must be one or the other.");
+    }
+    if ((includeUriChecker != null ^ excludeUriChecker != null) &&
+        (excludedPathPattern != null ^ includedPathPattern != null)) {
+      throw new RuntimeException("You cannot use both a path (withIncludedPathPattern/withExcludedPathPattern) and predicate based (withIncludedUriChecker/withExcludedUriChecker). Must be one or the other.");
+    }
```

Comment:
Should we use `IllegalStateException` here instead of `RuntimeException`?

Reply:
Done